src/segment-loader.js

```diff
@@ -203,8 +205,13 @@ export const timestampOffsetForSegment = ({
   currentTimeline,
   startOfSegment,
   buffered,
+  calculateTimestampOffsetForEachSegment,
   overrideCheck
 }) => {
+  if (calculateTimestampOffsetForEachSegment) {
+    return buffered.length ? buffered.end(buffered.length - 1) : startOfSegment;
```

Comment:
Since we're using the same ternary statement below, I wonder if it would make sense to refactor this into a function and call it here and below? It would be a couple more lines of code, but might be more clear.

Reply:
done